feat interactive v2.0

axif0 opened the pull request:
<!---
Thank you for your pull request! 🚀
-->

### Contributor checklist

<!-- Please replace the empty checkboxes [] below with checked ones [x] accordingly. -->

- [X] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch
- [X] I have tested my code with the `pytest` command as directed in the [testing section of the contributing guide](https://github.com/scribe-org/Scribe-Data/blob/main/CONTRIBUTING.md#testing)

---

### Description
Added test for interactive, get
Added total and list functionalities 
Updated command messages
Added prefilled language, data-types functionalities.  

![interactive2 1](https://github.com/user-attachments/assets/7f342a23-f368-43fa-be89-3c70d803eb76)

<!--
Describe briefly what your pull request proposes to change. Especially if you have more than one commit, it is helpful to give a summary of what your contribution is trying to solve.

Also, please describe shortly how you tested that your change actually works.
-->

### Related PR

<!--- Scribe-Data prefers that pull requests be related to already open issues. -->
<!--- If applicable, please link to the issue by replacing ISSUE_NUMBER with the appropriate number below. -->
<!--- Feel free to delete this section if this does not apply. -->

- [Closed PR #497 ](https://github.com/scribe-org/Scribe-Data/pull/497)

andrewtavis commented:
The work in #497 is included here, @axif0? :)

axif0 commented:
> The work in #497 is included here, @axif0? :)

Ya..

andrewtavis commented:
No need :) Is a great idea 😊 Will review over the weekend!

andrewtavis commented:
Functionality wise this is looking really great, @axif0, but can we have interactive mode be per command, rather than adding it all onto `get`? So it'd be `scribe-data total -i` for the total lexemes?

axif0 commented:
Modified `scribe-data get -i` and created `scribe-data total -i`.

Updated PyICU funtionality. Now user don't have to execute the command again.
![image](https://github.com/user-attachments/assets/73bf38e7-7761-48ef-9adf-07c194e00957)

axif0 commented on src/scribe_data/cli/get.py:
```diff
@@ -136,7 +142,14 @@ def get_data(
 
     elif data_type in {"emoji-keywords", "emoji_keywords"}:
-        generate_emoji(language=language, output_dir=output_dir)
+        if not check_if_pyicu_installed():
```
By adding this user don't have to execute the command again. 
![image](https://github.com/user-attachments/assets/cf26d035-fd18-4db2-af95-e5600aa66bb5)

But I don't know why I got `403 error` and `EOFError`

andrewtavis commented:
Generally things are looking good, @axif0, but I'm getting a 403 forbidden for the query from total interactive mode. Might just be the server. Could I also ask you to take a look at the [changes I made in my commit](https://github.com/scribe-org/Scribe-Data/pull/500/commits/b173d11ac84386e603119ae959734e3c50531aa5) and familiarize yourself with the comment formatting - i.e. own line comments are capitalized and punctuated, and inline comments are lower case and don't have a period :)

axif0 commented:
Awesome, I've reviewed the changes you mentioned. You're right, following the standard comment formatting is important. Thanks for the guidance - I appreciate you taking the time to ensure I'm aware of these best practices. I'll apply the capitalization, punctuation, and formatting standards you described in future.

andrewtavis reviewed (approved):
All's working now, @axif0 :) Likely just the Wikidata Query Service needing us to back off for a moment. Thanks so much for these improvements! This really has come along so well 😊
